File: backend/DAL/Entities/Danger.cs
namespace DAL.Entities;

public class Danger : Entity {
  public DangerType Type { get; set; }
  public string Description { get; set; }
  public double Lat { get; set; }
  public double Lon { get; set; }
  public string? AddressName { get; set; }
  public string? Title { get; set; }
  public Guid CategoryId { get; set; }
  public string? OtherCategory { get; set; }
  public DateTime? ActiveAt { get; set; }
  public DateTime? ResolvedAt { get; set; }

  public bool IsActive =>
    ActiveAt != null && ActiveAt <= DateTime.UtcNow &&
    (ResolvedAt == null || ResolvedAt > DateTime.UtcNow);

  // Forgein Keys
  public DangerCategory Category { get; set; }

  // Reverse Navigation
  public ICollection<DangerMessage> Messages { get; set; }
  public ICollection<DangerRequest> Requests { get; set; }
  public ICollection<Notification> Notifications { get; set; }
}

public enum DangerType {
  Permanent,
  Temporary
}

File: frontend/src/services/entities/Danger.ts
import { Notification } from "./Notification";
import { DangerRequest } from "./request/DangerRequest";
import { DangerMessage } from "./DangerMessage";
import { DangerCategory } from "./DangerCategory";


export interface Danger {
    id?: string;
    createdAt?: Date;
    updatedAt?: Date | undefined;
    type?: DangerType;
    description?: string | undefined;
    lat?: number;
    lon?: number;
    addressName?: string | undefined;
    title?: string | undefined;
    categoryId?: string;
    otherCategory?: string | undefined;
    activeAt?: Date | undefined;
    resolvedAt?: Date | undefined;
    readonly isActive?: boolean;
    category?: DangerCategory;
    messages?: DangerMessage[] | undefined;
    requests?: DangerRequest[] | undefined;
    notifications?: Notification[] | undefined;
}

export enum DangerType {
    Permanent = "Permanent",
    Temporary = "Temporary"
}

